Source/GSStoryboardTransform.m

```diff
@@ -0,0 +1,747 @@
+        }
+      if ([coder containsValueForKey: @"NSSender"])
+        {
+          [self setSender: [coder decodeObjectForKey: @"NSSender"]];
+        }
+      if ([coder containsValueForKey: @"NSIdentifier"])
+        {
+          [self setIdentifier: [coder decodeObjectForKey: @"NSIdentifier"]];
+        }
+      if ([coder containsValueForKey: @"NSKind"])
+        {
+          [self setKind: [coder decodeObjectForKey: @"NSKind"]];
+        }
+    }
+  return self;
+}
+
+- (void) encodeWithCoder: (NSCoder *)coder
+{
+  // this is never encoded directly...
+}
+@end
+
+@implementation NSControllerPlaceholder
+
+- (NSString *) storyboardName
+{
+  return _storyboardName;
+}
+
+- (void) setStoryboardName: (NSString *)name
+{
+  ASSIGNCOPY(_storyboardName, name);
+}
+
+- (id) copyWithZone: (NSZone *)z
+{
+  NSControllerPlaceholder *c = [[NSControllerPlaceholder allocWithZone: z] init];
+  [c setStoryboardName: _storyboardName];
+  return c;
+}
+
+- (instancetype) initWithCoder: (NSCoder *)coder
+{
+  self = [super init];
+  if ([coder allowsKeyedCoding])
+    {
+      if ([coder containsValueForKey: @"NSStoryboardName"])
+        {
+          [self setStoryboardName: [coder decodeObjectForKey: @"NSStoryboardName"]];
+        }
+    }
+  return self;
+}
+
+- (void) encodeWithCoder: (NSCoder *)coder
+{
+  // this is never encoded directly...
+}
+
+- (id) instantiate
+{
+  NSStoryboard *sb = [NSStoryboard storyboardWithName: _storyboardName
+                                               bundle: [NSBundle mainBundle]];
+  return [sb instantiateInitialController];
+}
+
+@end
+
+@implementation GSStoryboardTransform
+
+- (instancetype) initWithData: (NSData *)data
+{
+  self = [super init];
+  if (self != nil)
+    {
+      NSXMLDocument *xml = [[NSXMLDocument alloc] initWithData: data
+                                                       options: 0
+                                                         error: NULL];
+      
+      _scenesMap = [[NSMutableDictionary alloc] initWithCapacity: 10];
+      _controllerMap = [[NSMutableDictionary alloc] initWithCapacity: 10];
+      _documentsMap = [[NSMutableDictionary alloc] initWithCapacity: 10];
+      _identifierToSegueMap = [[NSMutableDictionary alloc] initWithCapacity: 10];
+      
+      [self processStoryboard: xml];
+      RELEASE(xml);
+    }
+  return self;
+}
+
+- (void) dealloc
+{
+  RELEASE(_initialViewControllerId);
+  RELEASE(_applicationSceneId);
+  RELEASE(_scenesMap);
+  RELEASE(_controllerMap);
+  RELEASE(_documentsMap);
+  RELEASE(_identifierToSegueMap);
+  [super dealloc];
+}
+
+- (NSString *) initialViewControllerId
+{
+  return _initialViewControllerId;
+}
+
+- (NSString *) applicationSceneId
+{
+  return _applicationSceneId;
+}
+
+- (NSDictionary *) scenesMap
+{
+  return _scenesMap;
+}
+
+- (NSDictionary *) controllerMap
+{
+  return _controllerMap;
+}
+
+- (NSDictionary *) documentsMap
+{
+  return _documentsMap;
+}
+
+- (NSDictionary *) identifierToSegueMap
```

Comment:
Is this ever used?

Reply:
Yes, it's used to get the segueMap to set on the controller in processSegues:forId: and subsequently the identifier is used to resolve the destination controller.

Reply:
I was referring to this method and this indeed never is used. The next method fills the role you are referring to.

Reply:
Old habit, I just wanted a way to get it if needed.  I will remove it.